src/js/libcs/OpSound.js

```diff
@@ -1,48 +1,457 @@
+            erg = evaluate(modifs.damp);
+            if (erg.ctype === 'number') {
+                damp = erg.value.real;
+            }
+        }
+        if (modifs.duration !== undefined) {
+            erg = evaluate(modifs.duration);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.stop !== undefined) {
+            erg = evaluate(modifs.stop);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.harmonics !== undefined) {
+            erg = evaluate(modifs.harmonics);
+            if (erg.ctype === 'list') {
+                harmonics = erg.value;
+            }
+        }
         if (modifs.line !== undefined) {
-
             erg = evaluate(modifs.line);
             if (erg.ctype === 'number') {
-                linenumber = Math.floor(erg.value.real);
-                if (linenumber < 0) {
-                    linenumber = 0;
-                }
-                if (linenumber > 10) {
-                    linenumber = 10;
-                }
+                var val = erg.value.real;
+            }
+            else if (erg.ctype === 'string'){
+              var val = erg.value;
+            }
+            else {
+                var val = 0;
+            }
+            if (val === 0) {
+              line = 0;
             }
+            else if (!linesDict[val]){
+              line = linesCounter;
+              linesDict[val] = line;
+              linesCounter++;
+              if (linesCounter > lines.length){
+                lines.push(0);
+              }
+            }
+            else {
+              line = linesDict[val];
+            }
+        }
+        if (modifs.partials !== undefined) {
+          erg = evaluate(modifs.partials);
+          if (erg.ctype === 'list') {
+              partials = erg.value;
+          }
+        }
+        if (modifs.restart !== undefined) {
+          erg = evaluate(modifs.restart);
+          if (erg.ctype === 'boolean') {
+              restart = erg.value;
+          }
         }
     }
 
+    var freq = evaluate(args[0]).value.real;
+    var amp = 0.5;
+    var damp = 0;
+    var duration = 1;
+    var harmonics = List.asList(CSNumber.real(1)).value;
+    var line = 0;
+    var partials = Array(harmonics.length).fill(CSNumber.real(1));
+    var attack = 0.01;
+    var release = 0.01;
+    var restart = true;
 
-    var v0 = evaluateAndVal(args[0]);
-    var linenumber = 0;
-    if (v0.ctype === 'number') {
-        handleModifs();
-        var lines = sound.lines;
-        var f = v0.value.real;
-        if (lines[linenumber] === 0) {
-            // Was bitte sollte die Funktion T an dieser Stelle sein?
-            // lines[linenumber]=T("sin", {freq:f,mul:0.6}).play();
+    handleModifs();
+    if (harmonics.length>partials.length){
+      partials = Array(harmonics.length).fill(CSNumber.real(1))
```

Comment:
missing semicolon. 
Some output "Ignoring modifier partials, because the length of partials does not match the length of harmonics" would be nice